mitosheet/src/components/taskpanes/SplitTextToColumns/SplitTextToColumnsTaskpane.tsx

```diff
@@ -0,0 +1,315 @@
+import Spacer from "../../spacing/Spacer";
+
+interface SplitTextToColumnsTaskpaneProps {
+    mitoAPI: MitoAPI;
+    analysisData: AnalysisData;
+    sheetDataArray: SheetData[];
+    selectedSheetIndex: number;
+    setUIState: React.Dispatch<React.SetStateAction<UIState>>;
+    dfNames: string[];
+    startingColumnID: ColumnID | undefined
+}
+
+interface SplitTextToColumnsResult {
+    num_cols_created: number;
+}
+
+const delimiters = {',': 'Comma', '-': 'Dash', '\t': 'Tab', ' ': 'Space'}
+
+const getDefaultParams  = (startingColumnID: ColumnID | undefined, sheetDataArray: SheetData[], sheetIndex: number): SplitTextToColumnsParams | undefined => {
+    if (sheetDataArray.length === 0 || sheetDataArray[sheetIndex] === undefined && sheetIndex > 0) {
+        return undefined;
+    }
+
+    let columnID = undefined
+    if (startingColumnID !== undefined && Object.keys(sheetDataArray[sheetIndex].columnIDsMap).includes(startingColumnID)) {
+        // If the startingColumnID exists in the sheet, then use it. 
+        columnID = startingColumnID
+    } else {
+        // Otherwise use the first column if there is a column. Otherwise, undefined
+        columnID = Object.keys(sheetDataArray[sheetIndex]?.columnIDsMap || {})[0]
+    }
+
+    return {
+        sheet_index: sheetIndex,
+        column_id: columnID,
+        delimiters: [], 
+        new_column_header_suffix: getNewColumnHeader()
+    }
+}
+
+/* 
+    This taskpane allows users to split a column into multiple columns 
+    by separating on a delimeter
+*/
+const SplitTextToColumnsTaskpane = (props: SplitTextToColumnsTaskpaneProps): JSX.Element => {
+
+    const {params, setParams, loading, edit, editApplied, result} = useSendEditOnClick<SplitTextToColumnsParams, SplitTextToColumnsResult>(
+        getDefaultParams(props.startingColumnID, props.sheetDataArray, props.selectedSheetIndex),
+        StepType.SplitTextToColumns, 
+        props.mitoAPI,
+        props.analysisData,
+    )
+
+    const [preview, setPreview] = useState<(string | number | boolean)[][] | undefined>([])
+    
+    // When the startingColumnID is updated outside of the taskpane, set it as the column getting split
+    useEffect(() => {
+        setParams(prevParams => {
+            const newParams = getDefaultParams(props.startingColumnID, props.sheetDataArray, props.selectedSheetIndex);
+            if (newParams) {
+                return newParams;
+            }
+            return prevParams;
+        });
+    }, [props.startingColumnID])
+
+    useEffect(() => {
+        if (params !== undefined && params.delimiters.length > 0) {
+            // If there is at least one delimiter, load the preview
+            void loadSplitTextToColumnsPreview()
+        } else {
+            setPreview(undefined)
+        }
+    }, [params])
+    
+
+    async function loadSplitTextToColumnsPreview() {
+
+        if (params !== undefined && params.column_id !== undefined) {
```

Comment:
It's weird you do this check in both the effect above and here. Why not do all the checks in the effect, or do all the checks in this function? Splitting them across these two is just confusing.